### ISSUE TITLE
CondFormats dependencies in DataFormats

### ISSUE THREAD

**Issue by davidlange6** [cms-sw/cmssw#30189]
I'd like to remove the dependency of DataFormats/PatCandidates on CondFormats/L1TObjects (which I believe are not supposed to be allowed...). 

This can be accomplished by moving two enums from CondFormats/L1TObjects/interface/L1GtDefinitions.h to some DataFormats package. (presumably DataFormats/L1TGlobal). Are there reasons not to do this or to do it differently?

**Comment by davidlange6** [comment]
assign l1

**Comment by davidlange6** [comment]
assign core

**Comment by smuzaffar** [comment]
sounds reasonable to me.

**Comment by makortel** [comment]
Sounds good to me too.

**Comment by guitargeek** [comment]
A related issue might be the L1Trigger dependency in DataFormats (it's just this single one):
https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TMuon/src/EMTFHit.cc#L2

Wouldn't the DataFormats have no dependencies in an ideal world, besides FWCore and maybe CommonTools for tests?

**Comment by makortel** [comment]
@guitargeek Good catch, that dependence is not allowed (any volunteers to move things around?).

The allowed dependencies on packages defining persisted data formats are listed here
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts#Restrictions_on_the_package_defi
(packages defining transient-only products have less constraints)

**Comment by guitargeek** [comment]
Thanks for the information. I don't know if it's good to move this code around without coordinating with the original developers, because it seems to be actively developed in the last years. 

However, if I were to do it I would just move all of these member functions implemented in `EMTFHit.cc` into some new file `EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFHitFunctions.h` (or something), because it's the only place where they are used. All of the member variables of `EMTFHit` are public, so it should be no problem to turn the member functions into free functions.

**Comment by davidlange6** [comment]
more fun are the CondFormats dependencies in DataFormats/RPCDigi. If one wants to make a class that is both storable in the event and storable in the conditions db, what is the design for that? (or is there one?)

**Comment by davidlange6** [comment]
I've proposed L1 changes to #31346

**Comment by davidlange6** [comment]
Ping about  DataFormats/RPCDigi.

**Comment by cecilecaillol** [comment]
Addressed in https://github.com/cms-sw/cmssw/commit/c13ac4554e31202e760920b0a3f63d31ed29b749

**Comment by cecilecaillol** [comment]
+l1
